src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_ProjectIndex.cs

```diff
@@ -52,12 +52,21 @@ private static async Task<ProjectIndex> CreateIndexAsync(Project project, Cancel
                 var namedTypes = new MultiDictionary<string, (DocumentId, DeclaredSymbolInfo)>(
                     project.Services.GetRequiredService<ISyntaxFactsService>().StringComparer);
 
+                var solutionKey = SolutionKey.ToSolutionKey(project.Solution);
+
+                var regularDocumentStates = project.State.DocumentStates;
+                var sourceGeneratorDocumentStates = await project.Solution.State.GetSourceGeneratedDocumentStatesAsync(project.State, cancellationToken).ConfigureAwait(false);
+
+                var allStates =
+                    regularDocumentStates.States.Select(kvp => (kvp.Key, kvp.Value)).Concat(
+                    sourceGeneratorDocumentStates.States.Select(kvp => (kvp.Key, (DocumentState)kvp.Value)));
```

Comment:
in an ideal future, we can change this entire pipeline to be streaming.  then, we only pay the price of SG docs after doing the normal search.